frontend/packages/console-app/src/components/tour/tour-context.ts

```diff
@@ -0,0 +1,86 @@
+import { createContext, useReducer, Reducer, Dispatch, ReducerAction } from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import { getFlagsObject } from '@console/internal/reducers/features';
+import { useExtensions } from '@console/plugin-sdk';
+import { RootState } from '@console/internal/redux';
+import { isGuidedTour, GuidedTour } from '@console/plugin-sdk/src/typings/guided-tour';
+import { TourActions } from './const';
+import { getTourLocalStorageForPerspective, filterTourBasedonPermissionAndFlag } from './utils';
+import { TourDataType } from './type';
+
+export const tourReducer = (state: TourState, action: TourActions) => {
```

Comment:
Your actions should only need to update the state it affects and spread the existing state into the return value.